### ISSUE TITLE
Forwarding broken in Torrent and Grid view bugged

### ISSUE THREAD

**Issue by OtakuVadER** [RepoDevil/Mr.Matagi#4]
https://github.com/RepoDevil/Dantotsu/assets/88473676/d2dcd703-c096-4541-a726-fd74115a5656
Clicking+85 resets the stream and same with double tapping
And also we cant forward video from seek bar
![Screenshot_2024-04-02-13-13-51-03_1ef8a8c8d362916fa05b876bc674d1c6](https://github.com/RepoDevil/Dantotsu/assets/88473676/d707f80f-fd44-49f0-9a8b-1cfdccab1e7e)
Another bug which i found grid view is fucked same with compact view👇👇
![Screenshot_2024-04-02-13-14-43-57_1ef8a8c8d362916fa05b876bc674d1c6](https://github.com/RepoDevil/Dantotsu/assets/88473676/e4360328-2909-4c84-a111-a263421da685)

**Comment by AbandonedCart** [comment]
Assuming the grid thing is another one of their bugs. I'll fix that first. I have a feeling fast forward is going to be another huge rewrite.

**Comment by AbandonedCart** [comment]
oddly, the +85 appears to be working normally on my end, but it does take some time to load if the video wasn't cached far enough ahead

**Comment by OtakuVadER** [comment]
> oddly, the +85 and double tap appears to be working normally on my end, but it does take some time to load if the video wasn't cached far enough ahead
> 
> I wonder if the particular file in that stream doesn't support fast forward. Playing partial avi files has that issue.

Well then let me try any other stream and let it cache more !!

**Comment by AbandonedCart** [comment]
If it works for others, let me know. I can at least add a warning for it

**Comment by OtakuVadER** [comment]
![IMG_20240402_234849](https://github.com/RepoDevil/Dantotsu/assets/88473676/5e89b99f-7bdc-42b4-8ed6-8098773e2b77)
Yea works for me too tried oshi no ko and mushoku tensei and your lie in April
But that specific server doesn't allow me to use +85 and double tap to forward

**Comment by AbandonedCart** [comment]
Looks like the default for when it can't find a position is the beginning of the video. I'm trying to find a way to make that the current position. 

TL;DR if it can't skip, it shouldn't try

**Comment by AbandonedCart** [comment]
@ni3x It's being tested here

**Comment by OtakuVadER** [comment]
> Looks like the default for when it can't find a position is the beginning of the video when a resume point isn't set. There is an old bug in ExoPlayer that setting seekTo(0) can cause the video to keep going back to the beginning when trying to seek later. I set the minimum seek to 1, which is the workaround they provided. Hopefully that fixes it.

No it doesn't fix that issue it's still there

https://github.com/RepoDevil/Dantotsu/assets/88473676/e7bbb4a6-9df2-420e-80d7-30fc974b62a6

**Comment by AbandonedCart** [comment]
In that case, it's definitely something with the video. I'm just not sure if there is a way to predict it, but I have some ideas.

**Comment by OtakuVadER** [comment]
> In that case, it's definitely something with the video. I'm just not sure if there is a way to predict it, but I have some ideas.

If it is possible to fix this then try those things or else i think there is nothing we can do anyways if you need help in any testing i am willing to help in testing!!

**Comment by AbandonedCart** [comment]
Sorry, I ended up fixing 100 other issues and never had a definitive solution for this. Ill try to get something put together today.

**Comment by OtakuVadER** [comment]
Yea that's okay
Iet me try this same stream in aniyomi and see if skipping works in it !

**Comment by OtakuVadER** [comment]
Well apparently that tenrai sensei stream is missing from everywhere so am unable to test it!!

**Comment by AbandonedCart** [comment]
Saw they are trying to add torrent support upstream. It will be interesting to see how bad of a train wreck that turns into because it's already headed in the wrong direction.

**Comment by OtakuVadER** [comment]
I saw it they made a module of it and they are going to make new download system in module too to reduce the application size

**Comment by AbandonedCart** [comment]
Unfortunately, that all comes at a cost to performance.

**Comment by AbandonedCart** [comment]
Lol I don't think anything will ever be perfect, but i'll be happy when I can look at it and not see things that *need* to be fixed.

**Comment by AbandonedCart** [comment]
I'm not sure I understand. Do you mean searching for something from the popup page?

**Comment by OtakuVadER** [comment]
Umm kind of like a blank entry
Like there is jujutsu kaisen an entry a page kind of wih information 
But one with no information 
But one which is blank and we can choose any movie source and play movie with wrong title 
But instead of wrong title 
There should be a search option

**Comment by AbandonedCart** [comment]
Oh ok. I'll see what I can do about that.

**Comment by OtakuVadER** [comment]
I also have another suggestion but that's related to novel so am opening new issue for that

**Comment by AbandonedCart** [comment]
Alright, sounds good.
